packages/rulesets/src/spectral/test/lro-delete-provisioning-state-specified.test.ts

```diff
@@ -0,0 +1,442 @@
+            type: "integer",
+            format: "int32",
+            minimum: 0,
+            maximum: 255,
+          },
+          id: {
+            type: "string",
+          },
+        },
+      },
+    },
+  }
+  return linter.run(oasDoc).then((results) => {
+    expect(results.length).toBe(1)
+    expect(results[0].path.join(".")).toBe("paths./foo.delete.responses.200")
+    expect(results[0].message).toContain(
+      "200 response schema in long running DELETE operation is missing ProvisioningState property. A LRO DELETE operations 200 response schema must have ProvisioningState specified."
+    )
+  })
+})
+
+test("ProvisioningStateSpecifiedForLRODelete should find no errors", () => {
+  const oasDoc = {
+    swagger: "2.0",
+    paths: {
+      "/foo": {
+        delete: {
+          tags: ["SampleTag"],
+          operationId: "Foo_Update",
+          description: "Test Description",
+          parameters: [
+            {
+              name: "foo_delete",
+              in: "body",
+              schema: {
+                $ref: "#/definitions/FooRequestParams",
+              },
+            },
+          ],
+          responses: {
+            "200": {
+              description: "Success",
+              schema: {
+                $ref: "#/definitions/FooRule",
+              },
+            },
+          },
+          "x-ms-long-running-operation": true,
+          "x-ms-long-running-operation-options": {
+            "final-state-via": "azure-async-operation",
+          },
+        },
+      },
+    },
+    definitions: {
+      FooRequestParams: {
+        allOf: [
+          {
+            $ref: "#/definitions/FooProps",
+          },
+        ],
+      },
+      FooProps: {
+        properties: {
+          servicePrecedence: {
+            description:
+              "A precedence value that is used to decide between services when identifying the QoS values to use for a particular SIM. A lower value means a higher priority. This value should be unique among all services configured in the mobile network.",
+            type: "integer",
+            format: "int32",
+            minimum: 0,
+            maximum: 255,
+          },
+          id: {
+            type: "string",
+          },
+        },
+      },
+      FooResource: {
+        "x-ms-azure-resource": true,
+        properties: {
+          provisioningState: {
+            type: "string",
+            enum: ["Creating", "Canceled", "Deleting", "Failed"],
+          },
+        },
+      },
+      FooRule: {
+        type: "object",
+        properties: {
+          properties: {
+            $ref: "#/definitions/FooResource",
+            "x-ms-client-flatten": true,
+          },
+        },
+        required: ["properties"],
+      },
+    },
+  }
+  return linter.run(oasDoc).then((results) => {
+    expect(results.length).toBe(0)
+  })
+})
+
+test("ProvisioningStateSpecifiedForRegularDelete should find no errors", () => {
```

Comment:
>test [](http://example.com/codeflow?start=0&length=4)

Couple more tests

1. Synchronous delete without provisioning state property should find no errors.
2. LRO delete with a properties property but no provisioningState property inside properties should find errors

Reply:
Add similar tests for patch and put as well

Reply:
I dont find the second test I called out. Can u doublecheck?

Reply:
pl check if the first test is implemented for all verbs

Reply:
last test in every test class should be testing 1st condition
"ProvisioningStateSpecifiedForSyncPut/Patch/Delete should find no errors"

Reply:
I saw that but that is when provisioning state is specified for sync put. #1 above is for when provisioningState is NOT specified for sync put.